src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
-import './App.css';
+import React, {useEffect} from 'react';
+import {
+  RouterProvider,
+} from "react-router-dom";
+import {useAppDispatch} from "./Hooks/hook";
+import auth from "./Firebase";
+import {userAuth, userLogOut, userToken} from "./Redux/Store/auth/authSlice";
+import axios from "axios";
+import {appRouter} from "./Routes/Router";
+
+
+function App():JSX.Element {
+  const dispatch = useAppDispatch()
+  useEffect(()=> {
```

Comment:
Not the best place to handle all of this, you can have some other component which can be responsible for this.

Reply:
I have modified the majority of the forms to work with Formik and Yup.

Reply:
Done! Now I have a custom hook that handles this functionality.